MonoGame.Framework/Graphics/Effect/Effect.cs

```diff
@@ -228,68 +191,111 @@ internal protected override void GraphicsDeviceResetting()
 
         #region Effect File Reader
 
-		private void ReadEffect (BinaryReaderEx reader)
-		{
-			// TODO: Maybe we should be reading in a string 
-			// table here to save some bytes in the file.
-
-			// Read in all the constant buffers.
-			var buffers = (int)reader.ReadByte ();
-			ConstantBuffers = new ConstantBuffer[buffers];
-			for (var c = 0; c < buffers; c++) 
-            {				
-				var name = reader.ReadString ();               
-
-				// Create the backing system memory buffer.
-				var sizeInBytes = (int)reader.ReadInt16 ();
-
-				// Read the parameter index values.
-				var parameters = new int[reader.ReadByte ()];
-				var offsets = new int[parameters.Length];
-				for (var i = 0; i < parameters.Length; i++) 
-                {
-					parameters [i] = (int)reader.ReadByte ();
-					offsets [i] = (int)reader.ReadUInt16 ();
-				}
+        struct MGFXHeader
+        {
+            /// <summary>
+            /// The MonoGame Effect file format header identifier ("MGFX"). 
+            /// </summary>
+            public static readonly int MGFXSignature = (BitConverter.IsLittleEndian) ? 0x5846474D : 0x4D474658;
 
-                var buffer = new ConstantBuffer(GraphicsDevice,
-				                                sizeInBytes,
-				                                parameters,
-				                                offsets,
-				                                name);
-                ConstantBuffers[c] = buffer;
+            /// <summary>
+            /// The current MonoGame Effect file format version.            
+            /// </summary>            
+            public const int MGFXVersion = 10;            
+
+            public int Signature;
+            public int Version;
+            public int Profile;
+            public int EffectKey;
+            public int HeaderSize;
+
+            public static MGFXHeader ReadHeader(byte[] effectCode, int index)
```

Comment:
Why is this part of a struct now?

Reply:
It's basically the constructor for that class.